interface-impl-amf/src/main/java/org/mule/amf/impl/model/ActionImpl.java

```diff
@@ -142,16 +142,17 @@ public Resource getResource() {
    * @return
    */
   private static Map<String, Parameter> loadResolvedUriParameters(final Resource resource, Operation operation) {
-    final Map<String, Parameter> operationUriParams;
+    final Map<String, Parameter> operationUriParams = new HashMap<>();
     if (operation.request() != null) {
-      operationUriParams = operation.request().uriParameters().stream()
-          .filter(p -> !VERSION.equals(p.name().value()))
-          .collect(toMap(p -> p.name().value(), ParameterImpl::new));
-    } else {
-      operationUriParams = new HashMap<>();
+      List<amf.client.model.domain.Parameter> collectedUriParams = operation.request().uriParameters().stream()
+          .filter(p -> !VERSION.equals(p.name().value())).collect(toList());
+      for (amf.client.model.domain.Parameter uriParam : collectedUriParams) {
```

Comment:
is this equals to .collect(toMap(p -> p.name().value(), ParameterImpl::new)); ?

Reply:
I omitted the use of streams here because the duplicates but I've just realized that I can make use of the merge function. I will update

Reply:
Updated